lbrynet/lbrynet_daemon/auth/util.py

```diff
@@ -0,0 +1,82 @@
+import base58
+import hmac
+import hashlib
+import yaml
+import os
+import logging
+
+log = logging.getLogger(__name__)
+
+API_KEY_NAME = "api"
+
+
+def sha(x):
+    h = hashlib.sha256(x).digest()
+    return base58.b58encode(h)
+
+
+def generate_key(x=None):
+    if x is None:
+        return sha(os.urandom(256))
+    else:
+        return sha(x)
+
+
+class APIKey(dict):
+    def __init__(self, key, name=None):
+        self.key = key if isinstance(key, str) else key['token']
+        self.name = name if name else hashlib.sha256(self.key).hexdigest()
+        self.expiration = None if isinstance(key, str) else key.get('expiration', None)
+        self.update({self.name: {'token': self.key, 'expiration': self.expiration}})
+
+    @classmethod
+    def new(cls, expiration=None, seed=None, name=None):
+        key_val = generate_key(seed)
+        key = {'token': key_val, 'expiration': expiration}
```

Comment:
this is confusing. is `key` the key, or is `key_val`. it seems like `key_val` is the actual key, and `key` is a combination of the key and the expiration. maybe rename `key`?

in fact, why do you pass the key and expiration to `APIKey` as a dict, and then pass the name separately. why not pass all three as plain variables. what's the point of the dict?